gtsam/slam/dataset.cpp

```diff
@@ -585,8 +593,13 @@ BetweenFactorPose3s parse3DFactors(const string& filename) {
       mgtsam.block<3, 3>(3, 0) = m.block<3, 3>(3, 0);  // off diagonal
 
       SharedNoiseModel model = noiseModel::Gaussian::Information(mgtsam);
+      auto R12 = Rot3::Quaternion(qw, qx, qy, qz);
+      if (corruptingNoise) {
+        R12 = R12.retract(sampler.sample());
```

Comment:
We can just do `Sampler sampler(corruptingNoise);` immediately before this line, right?